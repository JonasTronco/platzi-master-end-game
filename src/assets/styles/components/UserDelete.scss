@import '../main.scss';

.user-delete {
  @include colCenCen;
  text-align: center;

  &__icon {
    width: $space_x10;
    height: $space_x10;
    fill: $secondary_color;
  }

  &__text {
    margin: $space_x2 0px;
  }

  &__buttons {
    @include flexCenCen

  }

  button:first-child {
    margin-right: $space_x4;
  }

}