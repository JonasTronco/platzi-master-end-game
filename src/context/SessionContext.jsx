import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
