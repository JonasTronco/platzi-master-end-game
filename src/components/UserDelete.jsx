import WarningIcon from '../assets/icons/warning.svg';
import '../assets/styles/components/UserDelete.scss';

function UserDelete() {
  return (
    <div className="user-delete">
      <WarningIcon className="user-delete__icon" />
      <h2 className="user-delete__text">Are you sure you want to delete this user?</h2>
      <div className="user-delete__buttons">
        <button className="btn--positive" type="submit">Accept</button>
        <button className="btn--negative" type="submit">Cancel</button>
      </div>
    </div>
  );
}

export default UserDelete;