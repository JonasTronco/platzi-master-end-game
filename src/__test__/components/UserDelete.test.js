import UserDelete from '../../components/UserDelete';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('UserDelete Component', () => {
  test('Should render UserDelete component', () => {
    const component = create(
      <SessionProvider>
        <UserDelete />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
