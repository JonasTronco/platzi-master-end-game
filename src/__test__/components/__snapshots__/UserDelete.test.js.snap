// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDelete Component Should render UserDelete component 1`] = `
<div
  className="user-delete"
>
  <div
    className="user-delete__icon"
  />
  <h2
    className="user-delete__text"
  >
    Are you sure you want to delete this user?
  </h2>
  <div
    className="user-delete__buttons"
  >
    <button
      className="btn--positive"
      type="submit"
    >
      Confirm
    </button>
    <button
      className="btn--negative"
      type="submit"
    >
      Cancel
    </button>
  </div>
</div>
`;
